lib/rubocop/cop/lint/interpolation_check.rb

```diff
@@ -23,10 +23,10 @@ class InterpolationCheck < Base
               'Use double quoted strings if you need interpolation.'
 
         def on_str(node)
-          parent = node.parent
-          return if parent && (parent.dstr_type? || parent.regexp_type?)
+          return if string_or_regex?(node.parent)
           return unless /(?<!\\)#\{.*\}/.match?(node.source)
           return if heredoc?(node)
+          return unless node.loc.begin && node.loc.end
```

Comment:
You might want to add a comment here explaining the need for this or extract into a method with a descriptive name.

Reply:
Changed.